### PR TITLE
Commerce 9231

### PR DESCRIPTION
**Pull request by brianikim**
@luca-pellizzon ,

https://issues.liferay.com/browse/COMMERCE-9231

We need to return NULL if cpDefinition is not approved or published. Unless I am wrong, we only utilize the method assuming that the cpDefinition is published or approved.

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/util/CPDefinitionHelperImpl.java [hunk 1]
```diff
@@ -213,6 +213,13 @@ private CPDefinitionSearcher _getCPDefinitionSearcher(
 	private String _getFriendlyURL(long cProductId, ThemeDisplay themeDisplay)
 		throws PortalException {
 
+		CPDefinition cpDefinition =
+			_cpDefinitionLocalService.fetchCPDefinitionByCProductId(cProductId);
+
+		if (!cpDefinition.isApproved() || !cpDefinition.isPublished()) {
+			return null;
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/util/CPDefinitionHelperImpl.java:220]: You're just returning a null here, did you check that all places where this "result" is used are managing a "null" value?
You should either add a null check there, or returning an empty string here.
cc @andrea-ale-sbarra

**Review by luca-pellizzon — CHANGES_REQUESTED** [review]
Just a clarification on how the null value is managed, thanks.

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/util/CPDefinitionHelperImpl.java (hunk 1 above)
Comment (brianikim) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/util/CPDefinitionHelperImpl.java]: @luca-pellizzon ,

I resent the PR and introduced the fix in CPDefinitionAssetRenderer.getURLViewInContext() instead. This ensures we don't have to do the check in CPDefinitionHelper and allows us to keep the fix more contained. Also, the method already has null checks being managed in all the places where the "result" is used.

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

**Review by luca-pellizzon — APPROVED** [review]

**Review by andrea-ale-sbarra — APPROVED** [review]

**Comment by andrea-ale-sbarra** [comment]
I'll will forward this as soon as the solution is accepted in LPP

**Comment by riccardo-alberti** [comment]
@luca-pellizzon @andrea-ale-sbarra is this PR still needed? If not please close

**Comment by andrea-ale-sbarra** [comment]
ci:forward

**Pull request closed without merging**